### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by darkdef**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

Added methods to interface

Fix - if you categoryName equals with exists categoryName, throwed exception

Added new features:
- immutable for change category&locale
- adding multi categories by one function

Addes tests

### CODE REVIEW

**Review by terabytesoftw — APPROVED** [review]
It looks great.

File: src/Translator.php [hunk 1]
```diff
@@ -43,27 +43,26 @@ public function __construct(
...
+        }
         $this->categories[$category->getName()] = $category;
     }
 
-    /**
-     * Set the default locale.
-     *
-     * @param string $locale
-     */
+    public function addMultiCategorySource(array $categories): void
```
Comment (terabytesoftw) [inline src/Translator.php:54]: Sounds better:

```php
public function addMultipleCategorySource(array $categories): void
```

Comment (darkdef) [reply src/Translator.php]: yep

**Comment by darkdef** [comment]
Before merge - need correcting documentation

**Review by samdark — APPROVED** [review]

File: src/TranslatorInterface.php [hunk 2]
```diff
@@ -9,6 +9,28 @@
  */
 interface TranslatorInterface
 {
+    /**
+     * @param Category $category Add category.
```
Comment (roxblnfk) [inline src/TranslatorInterface.php:13]:
```suggestion
     * Add category.
```

File: src/TranslatorInterface.php [hunk 3]
```diff
@@ -9,6 +9,28 @@
...
+     */
+    public function addCategorySource(Category $category): void;
+
+    /**
+     * @param Category[] $categories Adding many categories.
```
Comment (roxblnfk) [inline src/TranslatorInterface.php:18]:
```suggestion
     * Adding many categories.
     *
     * @param Category[] $categories
```

File: src/TranslatorInterface.php [hunk 4]
```diff
@@ -9,6 +9,28 @@
...
+     */
+    public function addCategorySources(array $categories): void;
+
+    /**
+     * Set the default locale.
+     *
+     * @param string $locale
```
Comment (roxblnfk) [inline src/TranslatorInterface.php:25]:
```suggestion
```

**Comment by samdark** [comment]
👍

**Pull request merged**